client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -159,29 +181,137 @@ public boolean ping(int timeout) {
-        //This is just a placeholder
+        LOG.debug("Registering POJO: {}", clazz.getName());
+
         //Create a new POJOSerializer with static .serialize(object, columns) methods
+        List<POJOSerializer> serializers = new ArrayList<>();
+        Map<String, Method> getterMethods = new HashMap<>();
+
+        for (Method method: clazz.getMethods()) {//Clean up the method names
+            String methodName = method.getName();
+            if (methodName.startsWith("get") || methodName.startsWith("has")) {
+                methodName = methodName.substring(3).toLowerCase();
+                getterMethods.put(methodName, method);
+            } if (methodName.startsWith("is")) {
+                methodName = methodName.substring(2).toLowerCase();
+                getterMethods.put(methodName, method);
+            }
+        }
+        this.getterMethods.put(clazz, getterMethods);//Store the getter methods for later use
+
+        for (ClickHouseColumn column : schema.getColumns()) {
+            String columnName = column.getColumnName().toLowerCase().replace("_", "");
+            serializers.add((obj, stream) -> {
+                if (!getterMethods.containsKey(columnName)) {
+                    LOG.warn("No getter method found for column: {}", columnName);
+                    return;
+                }
+                Method getterMethod = this.getterMethods.get(clazz).get(columnName);
+                Object value = getterMethod.invoke(obj);
+                boolean hasDefaults = this.hasDefaults.get(clazz);
+
+                //Handle null values
+                if (value == null) {
+                    if (hasDefaults && !column.hasDefault()) {//Send this only if there is no default
+                        BinaryStreamUtils.writeNonNull(stream);
+                    }
+                    BinaryStreamUtils.writeNull(stream);//We send this regardless of default or nullable
+                    return;
+                }
+
+                //Handle default
+                if (hasDefaults) {
+                    BinaryStreamUtils.writeNonNull(stream);//Write 0
+                }
+
+                //Handle nullable
+                if (column.isNullable()) {
+                    BinaryStreamUtils.writeNonNull(stream);//Write 0
+                }
+
+                //Handle the different types
+                SerializerUtils.serializeData(stream, value, column);
+            });
+        }
+        this.serializers.put(clazz, serializers);
+        this.hasDefaults.put(clazz, schema.hasDefaults());
     }
 
     /**
      * Insert data into ClickHouse using a POJO
      */
-    public Future<InsertResponse> insert(String tableName,
-                                     List<Object> data,
-                                     InsertSettings settings,
-                                     List<ClickHouseColumn> columns) throws ClickHouseException, SocketException {
+    public InsertResponse insert(String tableName,
```

Comment:
I think this list of exceptions is confusing because user will need to understand each case when exception is thrown to handle properly. For example, can we retry after ExecutionException? 
I suggest to wrap them all under fewer exceptions with simple meanings. For example, NetworkException, ValidationException, ClientException.